### PR TITLE
Small refactoring

### PR DESCRIPTION
**Pull request by MBkkt**
### Purpose

So lazy...

### Related Information

- [ ] Design document: ...
- [ ] Bench PR: ...

### Testing

- [ ] This change is a trivial rework or code cleanup without any test coverage.
- [ ] This PR adds tests that were used to verify all changes.
- [ ] There are tests in an external testing repository: ...
- [ ] This change is already covered by existing tests.

### CODE REVIEW

File: include/yaclib/async/future.hpp [hunk 1]
```diff
@@ -78,33 +78,28 @@ class Future final {
    * Return const ref of \ref Result from \ref Future
    *
    * Assume Ready is true. This method is NOT thread-safe and can be called multiple
```
Comment (mkornaukhov03) [inline include/yaclib/async/future.hpp:80]: multiple times?

File: src/util/intrusive_list.cpp [hunk 2]
```diff
@@ -1,54 +1,48 @@
...
+bool List::Empty() const noexcept {
+  YACLIB_DEBUG((_head.next == nullptr) != (_tail == &_head), "list empty invariant is failed");
   return _head.next == nullptr;  // valid only for linear
 }
-template <typename T>
-T& List<T>::PopFront() noexcept {
-  assert(!Empty());
+
+Node& List::PopFront() noexcept {
+  YACLIB_DEBUG(Empty(), "");
```
Comment (kononovk) [inline src/util/intrusive_list.cpp:39]: write error message

File: src/util/atomic_event.cpp [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#include <yaclib/config.hpp>
+
```
Comment (kononovk) [inline src/util/atomic_event.cpp:1]: remove this stirng

File: src/fault/random/random_device.cpp [hunk 4]
```diff
@@ -1,4 +1,4 @@
-#include <yaclib/config.hpp>
+
```
Comment (kononovk) [inline src/fault/random/random_device.cpp:1]: delete this

File: src/fault/thread/thread.cpp [hunk 5]
```diff
@@ -1,4 +1,4 @@
-#include <yaclib/config.hpp>
+
```
Comment (kononovk) [inline src/fault/thread/thread.cpp:1]: delete this

File: src/executor/thread_factory.cpp [hunk 6]
```diff
@@ -232,19 +231,19 @@ class HeavyThreadFactory : public ThreadFactory {
...
+  detail::List _threads;
 };
 
 class DecoratorThreadFactory : public BaseFactory {
  protected:
   explicit DecoratorThreadFactory(IThreadFactoryPtr base) : _base{std::move(base)} {
   }
 
-  IThreadPtr Acquire(IFuncPtr func) final {
+  IThread* Acquire(IFuncPtr func) final {
```
Comment (kononovk) [inline src/executor/thread_factory.cpp:242]: Maybe ref ?

File: src/executor/inline.cpp [hunk 7]
```diff
@@ -1,4 +1,4 @@
-#include <yaclib/config.hpp>
+
```
Comment (kononovk) [inline src/executor/inline.cpp:1]: remove this

**Review by kononovk — APPROVED** [review]

**Pull request merged**